application/lib/view_model/base_screen_view_model.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:application/utils/colors.dart';
+import 'package:application/utils/styles/styles.dart';
+import 'package:application/utils/translation_en.dart';
+import 'package:application/view/screens/add_product_screen.dart';
+import 'package:application/view/screens/home_screen.dart';
+import 'package:application/view/screens/offers_screen.dart';
+import 'package:application/view/screens/products_screen.dart';
+import 'package:application/view/screens/settings_screen.dart';
+import 'package:flutter/material.dart';
+
+class BaseScreenViewModel extends ChangeNotifier {
+  final List<Widget> pages = [
+    const HomeScreen(),
+    const ProductsScreen(),
+    Container(),
+    const OffersScreen(),
+    const SettingsScreen(),
+  ];
+  int _selectedIndex = 0;
+
+  int get selectedIndex => _selectedIndex;
+
+  void selectTab(int index, BuildContext context) {
+    if (index == 2) {
+      _openBottomSheet(context);
+    } else {
+      _selectedIndex = index;
+      notifyListeners();
+    }
+  }
+
+  void _openBottomSheet(BuildContext context) {
+    showModalBottomSheet(
+      context: context,
+      isScrollControlled: true,
+      builder: (BuildContext context) {
+        return Padding(
```

Comment:
Ezt az egyéni widgetet jól ki lehet emelni. A viewmodelben nem szerencsés ilyen hosszú widget kód, mert alapvetően szeparálni akarjuk a UI kódokat és a logikai kódokat, ahogy a view és viewmodel is szeparálódik. Kérlek a widgets mappába kerüljön egy új widget a megfelelő névvel. Úgy ahogy pl a CustomCircularProgressIndicator widgeted is létrejött.